components/common-go/baseserver/server.go

```diff
@@ -43,6 +44,11 @@ func New(name string, opts ...Option) (*Server, error) {
 	server.httpMux = http.NewServeMux()
 	server.http = &http.Server{Handler: server.httpMux}
 
+	metricsErr := server.initializeMetrics()
+	if metricsErr != nil {
+		return nil, fmt.Errorf("failed to initialize metrics: %w", err)
+	}
```

Comment:
Is there any reason that you don't use `err`?
```suggestion
	err := server.initializeMetrics()
	if metricsErr != nil {
		return nil, fmt.Errorf("failed to initialize metrics: %w", err)
	}
```

Reply:
Old habit. Normally I'd use 
```
if err := server.initializeMetrics(); err != nil {
	...
}
```
Which makes err local, and prevents shadowing. But since that's for some reason discouraged in our style guide, to pass a stricter version (than we have currently enabled) gometalinter you'd need to assign different error variables as error shadowing is not allowed. It is for a good reason, the exact mistake I made here where I check the error for nil, but wrap the error of `err` rather than `metricsErr`. I've now renamed to `err` but that will shadow the variable.